tfrecorder/input_schema.py

```diff
@@ -0,0 +1,163 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Defines input types for TFRecorder's input schema."""
+
+from collections import OrderedDict
+from dataclasses import dataclass
+from typing import Any, Dict, List
+
+import frozendict
+import tensorflow as tf
+import tensorflow_transform as tft
+from tensorflow_transform.tf_metadata import dataset_metadata
+from tensorflow_transform.tf_metadata import schema_utils
+
+from tfrecorder import types
+
+@dataclass
+class SupportedType:
+  """Base type for TFRecorder Types."""
+  feature_spec: tf.io.FixedLenFeature
+  allowed_values: List[Any]
+
+@dataclass
+class ImageUriType(SupportedType):
+  """Supports image uri columns."""
+  feature_spec=tf.io.FixedLenFeature([], tf.string)
+  allowed_values=[]
+
+@dataclass
+class SplitKeyType(SupportedType):
+  """Supports split key columns."""
+  feature_spec=tf.io.FixedLenFeature([], tf.string)
+  allowed_values = ['TRAIN', 'VALIDATION', 'TEST', 'DISCARD']
+
+@dataclass
+class IntegerInputType(SupportedType):
+  """Supports integer columns."""
+  feature_spec=tf.io.FixedLenFeature([], tf.int64)
+  allowed_values=[]
+
+@dataclass
+class FloatInputType (SupportedType):
+  """Supports float columns."""
+  feature_spec=tf.io.FixedLenFeature([], tf.float64)
+  allowed_values=[]
+
+#TODO(mikebernico): Implement in preprocess_fn
+@dataclass
+class StringInputType(SupportedType):
+  """Supports string input columns."""
+  feature_spec=tf.io.FixedLenFeature([], tf.string)
+  allowed_values=[]
+
+@dataclass
+class IntegerLabelType(SupportedType):
+  """Supports integer labels."""
+  feature_spec=tf.io.FixedLenFeature([], tf.int64)
+  allowed_values=[]
+
+@dataclass
+class StringLabelType(SupportedType):
+  """Supports string labels."""
+  feature_spec=tf.io.FixedLenFeature([], tf.string)
+  allowed_values=[]
+
+@dataclass
+class ImageSupportStringType(SupportedType):
+  """Supports generated image bytestrings."""
+  feature_spec=tf.io.FixedLenFeature([], tf.string)
+  allowed_values=[]
+
+@dataclass
+class ImageSupportIntType(SupportedType):
+  """Supports generated image ints (height, width, channels)."""
+  feature_spec=tf.io.FixedLenFeature([], tf.int64)
+  allowed_values=[]
+
+class Schema:
+  """Defines a TFRecorder input schema."""
+  def __init__(self,
+               schema_map: OrderedDict
+               ) -> None:
+    """Defines TFRecorder input schema.
+
+    Args:
+      schema_map: An ordered dictionary that maps input columns to
+        TFRecorder supported types.
+    """
+    self.split_key = None
+    self.image_uri_key = None
+    self.label_key = None
+    self.input_schema_map = schema_map
+    self.pre_tft_schema_map = {}
+
+    for k, v in schema_map.items():
+      if v.__name__ == SplitKeyType.__name__:
```

Comment:
Is there a way to make the comparison not explicitly invoke the `__name__` attribute?

Reply:
Solved